htdocs/societe/card.php

```diff
@@ -406,6 +406,20 @@
 			setEventMessages($langs->trans("ErrorFieldRequired", $langs->transnoentities("No_Email")), null, 'errors');
 		}
 
+		if ($action == 'add') {
+			$thirdparty = GETPOST('name');
+			$sql = 'SELECT nom FROM '.MAIN_DB_PREFIX.'societe WHERE nom="'.$db->escape($thirdparty).'"';
```

Comment:
It is frequent that 2 companies has the same name. 
Currenlty this is allowed into Dolibar but we may add an option to forget this.
You can add a test here 
if (! empty($conf->global->THIRDPARTY_NAME_MUST_BE_UNIQUE))

Reply:
What about individual ?  How would you manage namesake of humain being ?
How would you manage multiple business units, parent and child society with sans name ?

Reply:
It is possible to add a filter to this, like 

`
if (! empty($conf->global->THIRDPARTY_NAME_MUST_BE_UNIQUE))
`